biojava-structure/src/main/java/org/biojava/nbio/structure/ChainImpl.java

```diff
@@ -592,8 +592,10 @@ public String getSeqResSequence(){
 					PolymerType.POLYNUCLEOTIDE_ONLY.contains(cc.getPolymerType())){
 				// an amino acid residue.. use for alignment
 				String oneLetter= ChemCompGroupFactory.getOneLetterCode(cc);
-				if ( oneLetter == null || oneLetter.isEmpty() || oneLetter.equals("?"))
+				// AB oneLetter.length() should be one. e.g. in 1EMA it is 3 and this makes mapping residue to sequence impossible.
+				if ( oneLetter == null || oneLetter.isEmpty() || oneLetter.equals("?") || oneLetter.length()!=1) {
```

Comment:
I don't think we should just change this in this way. We have documented this as a feature in the tutorial. Until the chemical component representation of chromophores has been fixed, let's add a utility method somewhere which masks groups with oneLetter length > 1 as X.